src/main/java/seedu/address/ui/AutoCompleteCommandBox.java

```diff
@@ -42,31 +52,11 @@
             "delete mentor", "delete team", "find participant", "find mentor", "find team", "leaderboard", "getTop",
             "score add", "score sub", "score set", "history", "undo", "redo", "import", "export", "help");
 
-    /**
-     * Represents a popup for user to select a command that is suggested.
-     */
-    private ContextMenu commandsPopup;
-
-    /**
-     * Constructs an AutoCompleteTextField that provide command suggestions as user
-     * types in command inputs.
-     */
-    public AutoCompleteTextField() {
+    public AutoCompleteCommandBox(CommandExecutor commandExecutor) {
         super();
-        this.addEventFilter(KeyEvent.KEY_PRESSED, event -> {
-            KeyCombination upCombo = new KeyCodeCombination(KeyCode.UP, KeyCombination.ALT_DOWN);
-            KeyCombination downCombo = new KeyCodeCombination(KeyCode.DOWN, KeyCombination.ALT_DOWN);
-
-            if (upCombo.match(event)) {
-                KeyEvent upEvent = new KeyEvent(KeyEvent.KEY_PRESSED, "", "", KeyCode.UP, false, false, true, false);
-                this.fireEvent(upEvent);
-            }
+        this.commandExecutor = commandExecutor;
+        this.setPromptText("What can I do for you?");
 
-            if (downCombo.match(event)) {
-                KeyEvent downEvent = new KeyEvent(KeyEvent.KEY_PRESSED, "", "", KeyCode.UP, false, false, true, false);
-                this.fireEvent(downEvent);
-            }
```

Comment:
Just curious, why does this not work?

Reply:
Becuase the event is thrown to every node including itself, then thrown repeatedly, so it results in stack overflow error.

Reply:
Ah I see thanks 👍